src/gui/input.cpp

```diff
@@ -144,13 +143,26 @@ QString InputConv::convertMouse(Qt::MouseButton bt, QEvent::Type type, Qt::Keybo
  *
  * see QKeyEvent
  */
-QString InputConv::convertKey(const QString& text, int k, Qt::KeyboardModifiers mod)
+QString InputConv::convertKey(const QString& in_text, int k, Qt::KeyboardModifiers mod)
 {
+	QString text = in_text;
+
 	if (specialKeys.contains(k)) {
 		return QString("<%1%2>").arg(modPrefix(mod)).arg(specialKeys.value(k));
 	} else if (text.isEmpty()) {
-		// This is a special key we can't handle
-		return QString();
+		// on macs, text is empty for ctrl+key combos
+		if (mod & ControlModifier) {
+			// ignore ctrl or ctrl+alt by itself
+			if (k == Key_Control || k == Key_Alt) {
+				return QString();
+			} else {
+				// key code will be the value of the char (hopefully)
+				text = QString(QChar(k));
+			}
+		} else {
+			// This is a special key we can't handle
+			return QString();
+		}
 	}
```

Comment:
Reversing the logic here helps make it more readable, and avoids the extra string var

```diff
diff --git a/src/gui/input.cpp b/src/gui/input.cpp
index 7d91c97..b2ece2c 100644
--- a/src/gui/input.cpp
+++ b/src/gui/input.cpp
@@ -143,12 +143,18 @@ QString InputConv::convertMouse(Qt::MouseButton bt, QEvent::Type type, Qt::Keybo
  *
  * see QKeyEvent
  */
-QString InputConv::convertKey(const QString& in_text, int k, Qt::KeyboardModifiers mod)
+QString InputConv::convertKey(const QString& text, int k, Qt::KeyboardModifiers mod)
 {
-       QString text = in_text;
-
        if (specialKeys.contains(k)) {
                return QString("<%1%2>").arg(modPrefix(mod)).arg(specialKeys.value(k));
+       }
+
+       QChar c;
+       // Escape < and backslash
+       if (text == "<") {
+               return QString("<%1%2>").arg(modPrefix(mod)).arg("lt");
+       } else if (text == "\\") {
+               return QString("<%1%2>").arg(modPrefix(mod)).arg("Bslash");
        } else if (text.isEmpty()) {
                // on macs, text is empty for ctrl+key combos
                if (mod & ControlModifier) {
@@ -157,23 +163,17 @@ QString InputConv::convertKey(const QString& in_text, int k, Qt::KeyboardModifie
                                return QString();
                        } else {
                                // key code will be the value of the char (hopefully)
-                               text = QString(QChar(k));
+                               c = QChar(k);
                        }
                } else {
                        // This is a special key we can't handle
                        return QString();
                }
+       } else {
+               // Key event compression is disabled, text has one char
+               c = text.at(0);
        }
 
-       // Escape < and backslash
-       if (text == "<") {
-               return QString("<%1%2>").arg(modPrefix(mod)).arg("lt");
-       }
-       if (text == "\\") {
-               return QString("<%1%2>").arg(modPrefix(mod)).arg("Bslash");
-       }
-
-       QChar c = text.at(0);
        // Remove SHIFT
```